plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerReports.groovy

```diff
@@ -220,4 +224,27 @@ class TowerReports {
         final prefix = reportKey.startsWith("**/") ? "" : "**/"
         return "glob:${prefix}${reportKey}"
     }
+
+    /**
+     * Publish any built-in reports that are enabled to the reports file.
+     */
+    void publishRuntimeReports() {
+        final config = session.config
+        final files = []
+
+        if( config.navigate('report.enabled') )
+            files << config.navigate('report.file', ReportObserver.DEF_FILE_NAME)
+
+        if( config.navigate('timeline.enabled') )
+            files << config.navigate('timeline.file', TimelineObserver.DEF_FILE_NAME)
+
+        if( config.navigate('trace.enabled') )
+            files << config.navigate('trace.file', TraceFileObserver.DEF_FILE_NAME)
+
+        if( config.navigate('dag.enabled') )
+            files << config.navigate('dag.file', GraphObserver.DEF_FILE_NAME)
+
+        for( def file : files )
+            filePublish( (file as Path).complete() )
```

Comment:
This seems overlapping with the logic in the LogsHandler

https://github.com/nextflow-io/nextflow/blob/3460-publish-nextflow-reports/plugins/nf-tower/src/main/io/seqera/tower/plugin/LogsHandler.groovy#L61-L70

Reply:
Why is the logs handler saving the timeline report but not the others?

Reply:
Because that was the requirement

Reply:
What requirement? 😆 

In any case, the logs handler is saving the timeline file whereas this PR is adding the built-in reports to the Tower reports list. So there is no overlap